specs/fault-proof/stage-one/anchor-state-registry.md

```diff
@@ -0,0 +1,224 @@
+  - [isGameRegistered](#isgameregistered)
+  - [isGameRespected](#isgamerespected)
+  - [isGameBlacklisted](#isgameblacklisted)
+  - [isGameRetired](#isgameretired)
+  - [isGameProper](#isgameproper)
+  - [getAnchorRoot](#getanchorroot)
+  - [anchors](#anchors)
+  - [tryUpdateAnchorState](#tryupdateanchorstate)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Overview
+
+The `AnchorStateRegistry` was designed as a registry where `DisputeGame` contracts could store and
+register their results so that these results could be used as the starting states for new
+`DisputeGame` instances. These starting states, called "anchor states", allow new `DisputeGame`
+contracts to use a newer starting state to bound the size of the execution trace for any given
+game.
+
+We are generally aiming to shift the `AnchorStateRegistry` to act as a unified source of truth for
+the validity of `DisputeGame` contracts and their corresponding root claims. This specification
+corresponds to the first iteration of the `AnchorStateRegistry` that will move us in this
+direction.
+
+## Definitions
+
+### Dispute Game
+
+> See [Fault Dispute Game](fault-dispute-game.md)
+
+A Dispute Game is a smart contract that makes a determination about the validity of some claim. In
+the context of the OP Stack, the claim is generally assumed to be a claim about the value of an
+output root at a given L2 block height.
+
+### Respected Game Type
+
+The `OptimismPortal` contract defines a "respected game type" which is the Dispute Game type that
+the portal allows to be used for the purpose of proving and finalizing withdrawals. This mechanism
+allows the system to use multiple game types simultaneously while still ensuring that the
+`OptimismPortal` contract only trusts respected games specifically.
+
+### Registered Game
+
+A Dispute Game is considered to be a **Registered Game** if the game contract was created by the
+system's `DisputeGameFactory` contract.
+
+### Respected Game
+
+A Dispute Game is considered to be a **Respected Game** if the game contract's game type is
+currently the `respectedGameType` defined by the `OptimismPortal` contract. Games that are not
+Respected Games cannot be used as an Anchor Game. See [Respected Game Type](#respected-game-type)
+for more information.
+
+### Blacklisted Game
+
+A Dispute Game is considered to be a **Blacklisted Game** if the game contract's address is marked
+as blacklisted inside of the `OptimismPortal` contract.
+
+### Retired Game
+
+A Dispute Game is considered to be a **Retired Game** if the game contract was created before the
+retirement timestamp (`respectedGameTypeUpdatedAt`) defined in the `OptimismPortal` contract.
+
+### Proper Game
+
+A Dispute Game is considered to be a **Proper Game** if all of the following are true:
+
+- Game is a Registered Game
+- Game is a Respected Game
+- Game is not a Blacklisted Game
+- Game is not a Retired Game
```

Comment:
Notably, this definition doesn't consider the outcome of the game in any way. That makes sense in many ways (ie you can prove a withdrawal against an in-progress game) but feels like a key concept is missing here (e.g. you can't finalise a withdrawal against a game that resolved as CHALLENGER_WINS and probably shouldn't be able to prove against it either).

This isn't necessarily wrong - it could be left up to the portal to decide if a game is valid based on its status, but it's notable that `tryUpdateAnchorState` has to make the same determination.  I wonder if there should be a `isValid` method that determines if a game is a proper game and resolved as Defender Wins.

Reply:
Yep, this is all defined in #517